ee/fe/modules/040-control-plane-manager/hooks/update_cluster_kubernetes_version.go

```diff
@@ -0,0 +1,167 @@
+/*
+Copyright 2021 Flant JSC
+Licensed under the Deckhouse Platform Enterprise Edition (EE) license. See https://github.com/deckhouse/deckhouse/blob/main/ee/LICENSE
+*/
+
+package hooks
+
+import (
+	"encoding/base64"
+	"fmt"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/flant/addon-operator/pkg/module_manager/go_hook"
+	"github.com/flant/addon-operator/sdk"
+	"github.com/flant/shell-operator/pkg/kube_events_manager/types"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/yaml"
+)
+
+const minimalKubernetesVersion = "1.21"
+
+func applyDeckhousePodFilter(obj *unstructured.Unstructured) (go_hook.FilterResult, error) {
+	var isReady bool
+
+	pod := &v1.Pod{}
+	err := sdk.FromUnstructured(obj, pod)
+	if err != nil {
+		return nil, fmt.Errorf("cannot parse pod object from unstructured: %v", err)
+	}
+	for _, cond := range pod.Status.Conditions {
+		if cond.Type == v1.PodReady && cond.Status == v1.ConditionTrue {
+			isReady = true
+			break
+		}
+	}
+	return isReady, nil
+}
+
+func applySecretFilter(unstructured *unstructured.Unstructured) (go_hook.FilterResult, error) {
+	secret := &v1.Secret{}
+
+	if err := sdk.FromUnstructured(unstructured, secret); err != nil {
+		return nil, err
+	}
+
+	clusterConfigData := secret.Data["cluster-configuration.yaml"]
+
+	var parsedClusterConfig map[string]interface{}
```

Comment:
Two thoughts here:
1. You probably do not need to load cluster configuration here because you do it with a global hook and save it to the global values.
2. If you are still willing to load it here, you need to use the function from dhctl like in the global hook to be sure that the config is valid https://github.com/deckhouse/deckhouse/blob/c390e26ff76435df1ebb8bb3a0330e7c42f1e702/global-hooks/discovery/cluster_configuration.go#L85

Reply:
i will use the function